### DISCUSSION TITLE
~models.not~

### DISCUSSION THREAD

**Discussion by deepimpactmir** [typst/typst#4050] [Q&A] [answered]
It seems that there is no `models.not`, can we define our own symbol modifier to have this in Typst? The equivalent in LaTeX is $\not\models$. I see that there is `divides.not` and even `tack.double.not`, which is close but the symbol for $\models$ has a shorter $=$.

**Comment by wrzian** [comment, marked as answer]
You can do this with the `cancel()` function: https://typst.app/docs/reference/math/cancel/

```typ
$models$ \
#let modelsnot = $cancel(models, length: #90%)$
$modelsnot$
```
![image](https://github.com/typst/typst/assets/133046678/f7a1924c-ab18-4b34-b23a-87aaa98bc3be)

**Comment by Enivex** [comment]
> It seems that there is no `models.not`, can we define our own symbol modifier to have this in Typst? The equivalent in LaTeX is $\not\models$. I see that there is `divides.not` and even `tack.double.not`, which is close but the symbol for $\models$ has a shorter $=$.

Unfortunately there is no such symbol in unicode

**Reply by UnlimitedCookies** [reply]
There's `⊭` (U+22AD).

**Reply by Enivex** [reply]
> There's `⊭`.

That's a different symbol

**Reply by UnlimitedCookies** [reply]
According to [this Wikipedia article](https://en.wikipedia.org/wiki/Double_turnstile) and [this Wikipedia list](https://en.wikipedia.org/wiki/List_of_logic_symbols) this would be exactly to use for _Not models_ in logic contexts.

**Reply by Enivex** [reply]
> According to [this Wikipedia article](https://en.wikipedia.org/wiki/Double_turnstile) and [this Wikipedia list](https://en.wikipedia.org/wiki/List_of_logic_symbols) this would be exactly to use for _Not models_ in logic contexts.

⊧ is U+22A7 (Models)
⊨ is U+22A8 (True)
⊭ is U+22AD (Not True)

As you can see, 22AD is the negated form of 22A8, but there is no negated form of 22A7. Of course, you can use 22A8 and 22AD to mean _models_ and _models not_, but OP is specifically asking about the shorter 22A7.

**Comment by deepimpactmir** [comment]
Thanks for the quick response!
